src/egon/data/datasets/electrical_neighbours.py

```diff
@@ -940,6 +953,75 @@ def insert_generators(capacities):
         session.add(entry)
         session.commit()
 
+    # assign generators time-series data
+    map_carriers = {
+        "onwind": "wind_onshore",
+        "offwind-ac": "wind_offshore",
+        "offwind-dc": "wind_offshore",
+        "solar": "solar",
+    }
+
+    sql = f"""SELECT * FROM
+    {targets['generators_timeseries']['schema']}.
+    {targets['generators_timeseries']['table']}
+    WHERE scn_name = 'eGon100RE'
+    """
+    series_egon100 = pd.read_sql_query(sql, db.engine())
+
+    sql = f""" SELECT * FROM
+    {targets['generators']['schema']}.{targets['generators']['table']}
+    WHERE bus IN (
+        SELECT bus_id FROM
+                {targets['buses']['schema']}.{targets['buses']['table']}
+                WHERE country != 'DE'
+                AND scn_name = 'eGon2035')
+        AND scn_name = 'eGon2035'
+    """
+    gen_2035 = pd.read_sql_query(sql, db.engine())
+    gen_2035 = gen_2035[gen_2035.carrier.isin(map_carriers.values())]
+
+    sql = f""" SELECT * FROM
+    {targets['generators']['schema']}.{targets['generators']['table']}
+    WHERE bus IN (
+        SELECT bus_id FROM
+                {targets['buses']['schema']}.{targets['buses']['table']}
+                WHERE country != 'DE'
+                AND scn_name = 'eGon100RE')
+        AND scn_name = 'eGon100RE'
+    """
+    gen_100 = pd.read_sql_query(sql, db.engine())
+
+    gen_100["carrier"] = gen_100["carrier"].map(map_carriers)
+    gen_100 = gen_100[gen_100["carrier"].notna()]
+
+    # egon_2035_to_100 map the timeseries used in the scenario eGon100RE
+    # to the same bus and carrier for the scenario egon2035
+    egon_2035_to_100 = {}
+    for i, gen in gen_2035.iterrows():
+        gen_id_100 = gen_100[
+            (gen_100["bus"] == gen["bus"])
```

Comment:
When I understand these lines correctly, foreign buses must have the same index for both scenarios. Did you checked if that's always the case?

Reply:
@ClaraBuettner yes, you understood it correctly. I just did the verification and this is also the case for the calculations when using data from the whole country.